JASP-Engine/JASP/R/commonerrorcheck.R

```diff
@@ -91,12 +91,25 @@
       message <- paste0(openingMsg, '<ul>', message)
     }
   }
+
+  # See if we should add the variables causing the error (so far only for perfect correlations).
+  if (type == "varCovData" && ! is.null(args$variables) && length(args$variables) > 0) {
+    message <- paste0(message, "<ul><li> Note: The following pairs of variables: ")
+    for (i in 1:length(args$variables)) {
```

Comment:
maybe:

    nPair <- length(args$variables)
    if (nPairs > 10) {
      msg<- sprintf("<ul><li> Note: There were %d pairs of variables perfectly correlated. The first 10 are: ", nPairs)
    } else {
      msg<- "<ul><li> Note: The following pairs of variables are perfectly correlated: "
    }
    message <- paste0(message, msg)
    for (i in seq_len(nPairs)) {
      ...
    }

that way, if somebody drags in many, many bad variables, jasp won't accidentally crash because of it.